extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcProvider.java

```diff
@@ -0,0 +1,260 @@
+package io.quarkus.oidc.runtime;
+
+import java.security.Key;
+import java.time.Duration;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.jboss.logging.Logger;
+import org.jose4j.jwa.AlgorithmConstraints;
+import org.jose4j.jws.JsonWebSignature;
+import org.jose4j.jwt.consumer.ErrorCodeValidator;
+import org.jose4j.jwt.consumer.InvalidJwtException;
+import org.jose4j.jwt.consumer.JwtConsumer;
+import org.jose4j.jwt.consumer.JwtConsumerBuilder;
+import org.jose4j.jwx.JsonWebStructure;
+import org.jose4j.keys.resolvers.VerificationKeyResolver;
+import org.jose4j.lang.UnresolvableKeyException;
+
+import io.quarkus.oidc.AuthorizationCodeTokens;
+import io.quarkus.oidc.OIDCException;
+import io.quarkus.oidc.OidcConfigurationMetadata;
+import io.quarkus.oidc.OidcTenantConfig;
+import io.quarkus.oidc.common.runtime.OidcConstants;
+import io.quarkus.security.AuthenticationFailedException;
+import io.quarkus.security.identity.request.TokenAuthenticationRequest;
+import io.smallrye.jwt.algorithm.SignatureAlgorithm;
+import io.smallrye.jwt.util.KeyUtils;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.subscription.UniEmitter;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+
+public class OidcProvider {
+
+    private static final Logger LOG = Logger.getLogger(OidcProvider.class);
+
+    private final OidcProviderClient client;
+    private final RefreshableVerificationKeyResolver keyResolver;
+    private final OidcTenantConfig oidcConfig;
+
+    public OidcProvider(OidcProviderClient client, OidcTenantConfig oidcConfig, JsonWebKeyCache jwks) {
+        this.client = client;
+        this.oidcConfig = oidcConfig;
+        this.keyResolver = jwks == null ? null : new JsonWebKeyResolver(jwks, oidcConfig.token.forcedJwkRefreshInterval);
+    }
+
+    public OidcProvider(String publicKeyEnc, OidcTenantConfig oidcConfig) {
+        this.client = null;
+        this.oidcConfig = oidcConfig;
+        this.keyResolver = new LocalPublicKeyResolver(publicKeyEnc);
+    }
+
+    public Uni<TokenVerificationResult> verifyJwtToken(String token) {
+        JwtConsumerBuilder builder = new JwtConsumerBuilder();
```

Comment:
That's synchronous, just return the TokenVerificationResult, not wrapped in a Uni